### PR TITLE
fix: [cds-46460]: use clusterref if name doesn't exist

### PR DESCRIPTION
**Pull request by cjlee01**
### Summary

Use cluster ref for the label names in the specify cluster dropdown if the name doesn't exist 

#### Screenshots

BEFORE:

https://user-images.githubusercontent.com/92757601/201989918-4b86a85d-a962-4a36-b6a9-3cd1b95a42ae.mov

after:

https://user-images.githubusercontent.com/92757601/201990220-5f5fa6fb-cb0e-4b41-94f9-04dd85b6e39d.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/DeployClusterEntityInputStep.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default function DeployClusterEntityInputStep({
     /* istanbul ignore else */
     if (!isNil(clustersList)) {
       return clustersList.map(clusterInList => ({
-        label: clusterInList.name,
+        label: clusterInList.name || clusterInList.clusterRef,
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/DeployClusterEntityInputStep.tsx:82]: This should happen not here

**Review by AshwinBhatkal-Harness — CHANGES_REQUESTED** [review]
Please make the change in the useGetClustersData hook instead. 
Also, the bug tells ng ui merge step. So, why are we making the change here?

File: src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/DeployClusterEntityInputStep.tsx (hunk 1 above)
Comment (cjlee01) [reply src/modules/75-cd/components/PipelineSteps/DeployClusterEntityStep/DeployClusterEntityInputStep.tsx]: fixed this as per comment @AshwinBhatkal-Harness

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployClusters/DeployClusters.tsx [hunk 2]
```diff
@@ -81,7 +81,7 @@ function DeployClusters({
       setClustersSelectOptions([
         { label: getString('cd.pipelineSteps.environmentTab.allClustersSelected'), value: getString('all') },
         ...clusters.map(cluster => {
-          return { label: defaultTo(cluster.clusterRef, ''), value: defaultTo(cluster.clusterRef, '') }
+          return { label: defaultTo(cluster.name, ''), value: defaultTo(cluster.clusterRef, '') }
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployClusters/DeployClusters.tsx:84]: Shall we do cluster.name || cluster.clusterRef ?

Comment (cjlee01) [reply src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployClusters/DeployClusters.tsx:84]: @AshwinBhatkal-Harness what do you think?

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**